Sources/XMLCoder/Box/Box.swift

```diff
@@ -0,0 +1,238 @@
+//
+//  Box.swift
+//  XMLCoder
+//
+//  Created by Vincent Esche on 12/17/18.
+//
+
+import Foundation
+
+internal enum Box {
+    case null(NullBox)
+    case bool(BoolBox)
+    case decimal(DecimalBox)
+    case signedInteger(SignedIntegerBox)
+    case unsignedInteger(UnsignedIntegerBox)
+    case floatingPoint(FloatingPointBox)
+    case string(StringBox)
+    case array(ArrayBox)
+    case dictionary(DictionaryBox)
+    
+    var isNull: Bool {
+        guard case .null(_) = self else {
+            return false
+        }
+        return true
+    }
+    
+    var isFragment: Bool {
+        switch self {
```

Comment:
`isFragment` could probably also be a property on a new `Box` protocol that would help to avoid this `switch`